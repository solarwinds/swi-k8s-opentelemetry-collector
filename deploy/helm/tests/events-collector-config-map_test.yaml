# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test for events-collector-config-map
templates:
  - events-collector-config-map.yaml
tests:
  - it: Events config should match snapshot when using default values
    template: events-collector-config-map.yaml
    asserts:
      - matchSnapshot:
          path: data