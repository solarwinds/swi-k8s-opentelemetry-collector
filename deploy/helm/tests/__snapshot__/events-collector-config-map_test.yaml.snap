Events config should match snapshot when using default values:
  1: |
    events.config: |-
      exporters:
        otlp:
          endpoint: ${OTEL_ENVOY_ADDRESS}
          headers:
            Authorization: Bearer ${SOLARWINDS_API_TOKEN}
          retry_on_failure:
            enabled: true
            initial_interval: 10s
            max_elapsed_time: 300s
            max_interval: 30s
          sending_queue:
            enabled: true
            num_consumers: 10
            queue_size: 1000
          tls:
            insecure: ${OTEL_ENVOY_ADDRESS_TLS_INSECURE}
      extensions:
        health_check: {}
        memory_ballast:
          size_mib: 300
      processors:
        batch:
          send_batch_max_size: 512
          send_batch_size: 512
          timeout: 1s
        k8sattributes:
          auth_type: serviceAccount
          cronjob:
            association:
            - sources:
              - from: resource_attribute
                name: k8s.cronjob.name
              - from: resource_attribute
                name: k8s.namespace.name
            extract:
              labels:
              - from: cronjob
                key_regex: (.*)
                tag_name: k8s.cronjob.labels.$$1
          daemonset:
            association:
            - sources:
              - from: resource_attribute
                name: k8s.daemonset.name
              - from: resource_attribute
                name: k8s.namespace.name
            extract:
              labels:
              - from: daemonset
                key_regex: (.*)
                tag_name: k8s.daemonset.labels.$$1
          deployment:
            association:
            - sources:
              - from: resource_attribute
                name: k8s.deployment.name
              - from: resource_attribute
                name: k8s.namespace.name
            extract:
              labels:
              - from: deployment
                key_regex: (.*)
                tag_name: k8s.deployment.labels.$$1
          extract:
            labels:
            - from: pod
              key_regex: (.*)
              tag_name: k8s.pod.labels.$$1
            - from: namespace
              key_regex: (.*)
              tag_name: k8s.namespace.labels.$$1
            metadata:
            - k8s.deployment.name
            - k8s.replicaset.name
            - k8s.daemonset.name
            - k8s.job.name
            - k8s.cronjob.name
            - k8s.statefulset.name
            - k8s.node.name
          job:
            association:
            - sources:
              - from: resource_attribute
                name: k8s.job.name
              - from: resource_attribute
                name: k8s.namespace.name
            extract:
              labels:
              - from: job
                key_regex: (.*)
                tag_name: k8s.job.labels.$$1
          node:
            association:
            - sources:
              - from: resource_attribute
                name: k8s.node.name
            extract:
              labels:
              - from: node
                key_regex: (.*)
                tag_name: k8s.node.labels.$$1
          passthrough: false
          persistentvolume:
            association:
            - sources:
              - from: resource_attribute
                name: k8s.persistentvolume.name
            extract:
              labels:
              - from: persistentvolume
                key_regex: (.*)
                tag_name: k8s.persistentvolume.labels.$$1
          persistentvolumeclaim:
            association:
            - sources:
              - from: resource_attribute
                name: k8s.persistentvolumeclaim.name
              - from: resource_attribute
                name: k8s.namespace.name
            extract:
              labels:
              - from: persistentvolumeclaim
                key_regex: (.*)
                tag_name: k8s.persistentvolumeclaim.labels.$$1
          pod_association:
          - sources:
            - from: resource_attribute
              name: k8s.pod.name
            - from: resource_attribute
              name: k8s.namespace.name
          replicaset:
            association:
            - sources:
              - from: resource_attribute
                name: k8s.replicaset.name
              - from: resource_attribute
                name: k8s.namespace.name
            extract:
              labels:
              - from: replicaset
                key_regex: (.*)
                tag_name: k8s.replicaset.labels.$$1
          service:
            association:
            - sources:
              - from: resource_attribute
                name: k8s.service.name
              - from: resource_attribute
                name: k8s.namespace.name
            extract:
              labels:
              - from: service
                key_regex: (.*)
                tag_name: k8s.service.labels.$$1
          set_object_existence: true
          statefulset:
            association:
            - sources:
              - from: resource_attribute
                name: k8s.statefulset.name
              - from: resource_attribute
                name: k8s.namespace.name
            extract:
              labels:
              - from: statefulset
                key_regex: (.*)
                tag_name: k8s.statefulset.labels.$$1
        memory_limiter:
          check_interval: 1s
          limit_mib: 512
          spike_limit_mib: 128
        resource/events:
          attributes:
          - action: insert
            key: sw.k8s.agent.manifest.version
            value: ${MANIFEST_VERSION}
          - action: insert
            key: sw.k8s.agent.app.version
            value: ${APP_VERSION}
          - action: insert
            key: sw.k8s.cluster.uid
            value: ${CLUSTER_UID}
          - action: insert
            key: k8s.cluster.name
            value: ${CLUSTER_NAME}
          - action: insert
            key: sw.k8s.log.type
            value: event
        transform/cleanup_attributes_for_nonexisting_entities:
          log_statements:
          - context: log
            statements:
            - delete_key(resource.attributes, "k8s.pod.name") where resource.attributes["sw.k8s.pod.found"]
              == "false"
            - delete_key(resource.attributes, "k8s.deployment.name") where resource.attributes["sw.k8s.deployment.found"]
              == "false"
            - delete_key(resource.attributes, "k8s.statefulset.name") where resource.attributes["sw.k8s.statefulset.found"]
              == "false"
            - delete_key(resource.attributes, "k8s.replicaset.name") where resource.attributes["sw.k8s.replicaset.found"]
              == "false"
            - delete_key(resource.attributes, "k8s.daemonset.name") where resource.attributes["sw.k8s.daemonset.found"]
              == "false"
            - delete_key(resource.attributes, "k8s.job.name") where resource.attributes["sw.k8s.job.found"]
              == "false"
            - delete_key(resource.attributes, "k8s.cronjob.name") where resource.attributes["sw.k8s.cronjob.found"]
              == "false"
            - delete_key(resource.attributes, "k8s.node.name") where resource.attributes["sw.k8s.node.found"]
              == "false"
            - delete_key(resource.attributes, "sw.k8s.pod.found")
            - delete_key(resource.attributes, "sw.k8s.deployment.found")
            - delete_key(resource.attributes, "sw.k8s.statefulset.found")
            - delete_key(resource.attributes, "sw.k8s.replicaset.found")
            - delete_key(resource.attributes, "sw.k8s.daemonset.found")
            - delete_key(resource.attributes, "sw.k8s.job.found")
            - delete_key(resource.attributes, "sw.k8s.cronjob.found")
            - delete_key(resource.attributes, "sw.k8s.node.found")
        transform/entity_attributes:
          log_statements:
          - context: log
            statements:
            - set(resource.attributes["k8s.pod.name"], resource.attributes["k8s.object.name"])
              where resource.attributes["k8s.object.kind"] == "Pod"
            - set(resource.attributes["k8s.deployment.name"], resource.attributes["k8s.object.name"])
              where resource.attributes["k8s.object.kind"] == "Deployment"
            - set(resource.attributes["k8s.statefulset.name"], resource.attributes["k8s.object.name"])
              where resource.attributes["k8s.object.kind"] == "StatefulSet"
            - set(resource.attributes["k8s.replicaset.name"], resource.attributes["k8s.object.name"])
              where resource.attributes["k8s.object.kind"] == "ReplicaSet"
            - set(resource.attributes["k8s.daemonset.name"], resource.attributes["k8s.object.name"])
              where resource.attributes["k8s.object.kind"] == "DaemonSet"
            - set(resource.attributes["k8s.job.name"], resource.attributes["k8s.object.name"])
              where resource.attributes["k8s.object.kind"] == "Job"
            - set(resource.attributes["k8s.cronjob.name"], resource.attributes["k8s.object.name"])
              where resource.attributes["k8s.object.kind"] == "CronJob"
            - set(resource.attributes["k8s.node.name"], resource.attributes["k8s.object.name"])
              where resource.attributes["k8s.object.kind"] == "Node"
            - set(resource.attributes["k8s.namespace.name"], attributes["k8s.namespace.name"])
              where attributes["k8s.namespace.name"] != nil
            - delete_key(attributes, "k8s.namespace.name") where attributes["k8s.namespace.name"]
              != nil
        transform/namespace:
          log_statements:
          - context: log
            statements:
            - set(attributes["sw.namespace"], "sw.events.inframon.k8s")
        transform/severity:
          log_statements:
          - context: log
            statements:
            - set(severity_text, "Error") where attributes["k8s.event.reason"] == "Failed"
              or attributes["k8s.event.reason"] == "BackOff" or attributes["k8s.event.reason"]
              == "FailedKillPod" or attributes["k8s.event.reason"] == "FailedCreatePodContainer"
              or attributes["k8s.event.reason"] == "NetworkNotReady" or attributes["k8s.event.reason"]
              == "InspectFailed" or attributes["k8s.event.reason"] == "ErrImageNeverPull"
              or attributes["k8s.event.reason"] == "NodeNotReady" or attributes["k8s.event.reason"]
              == "NodeNotSchedulable" or attributes["k8s.event.reason"] == "KubeletSetupFailed"
              or attributes["k8s.event.reason"] == "FailedAttachVolume" or attributes["k8s.event.reason"]
              == "FailedMount" or attributes["k8s.event.reason"] == "VolumeResizeFailed"
              or attributes["k8s.event.reason"] == "FileSystemResizeFailed" or attributes["k8s.event.reason"]
              == "FailedMapVolume" or attributes["k8s.event.reason"] == "ContainerGCFailed"
              or attributes["k8s.event.reason"] == "ImageGCFailed" or attributes["k8s.event.reason"]
              == "FailedNodeAllocatableEnforcement" or attributes["k8s.event.reason"] ==
              "FailedCreatePodSandBox" or attributes["k8s.event.reason"] == "FailedPodSandBoxStatus"
              or attributes["k8s.event.reason"] == "FailedMountOnFilesystemMismatch" or
              attributes["k8s.event.reason"] == "InvalidDiskCapacity" or attributes["k8s.event.reason"]
              == "FreeDiskSpaceFailed" or attributes["k8s.event.reason"] == "FailedSync"
              or attributes["k8s.event.reason"] == "FailedValidation" or attributes["k8s.event.reason"]
              == "FailedPostStartHook" or attributes["k8s.event.reason"] == "FailedPreStopHook"
            - set(severity_number, 17) where attributes["k8s.event.reason"] == "Failed"
              or attributes["k8s.event.reason"] == "BackOff" or attributes["k8s.event.reason"]
              == "FailedKillPod" or attributes["k8s.event.reason"] == "FailedCreatePodContainer"
              or attributes["k8s.event.reason"] == "NetworkNotReady" or attributes["k8s.event.reason"]
              == "InspectFailed" or attributes["k8s.event.reason"] == "ErrImageNeverPull"
              or attributes["k8s.event.reason"] == "NodeNotReady" or attributes["k8s.event.reason"]
              == "NodeNotSchedulable" or attributes["k8s.event.reason"] == "KubeletSetupFailed"
              or attributes["k8s.event.reason"] == "FailedAttachVolume" or attributes["k8s.event.reason"]
              == "FailedMount" or attributes["k8s.event.reason"] == "VolumeResizeFailed"
              or attributes["k8s.event.reason"] == "FileSystemResizeFailed" or attributes["k8s.event.reason"]
              == "FailedMapVolume" or attributes["k8s.event.reason"] == "ContainerGCFailed"
              or attributes["k8s.event.reason"] == "ImageGCFailed" or attributes["k8s.event.reason"]
              == "FailedNodeAllocatableEnforcement" or attributes["k8s.event.reason"] ==
              "FailedCreatePodSandBox" or attributes["k8s.event.reason"] == "FailedPodSandBoxStatus"
              or attributes["k8s.event.reason"] == "FailedMountOnFilesystemMismatch" or
              attributes["k8s.event.reason"] == "InvalidDiskCapacity" or attributes["k8s.event.reason"]
              == "FreeDiskSpaceFailed" or attributes["k8s.event.reason"] == "FailedSync"
              or attributes["k8s.event.reason"] == "FailedValidation" or attributes["k8s.event.reason"]
              == "FailedPostStartHook" or attributes["k8s.event.reason"] == "FailedPreStopHook"
            - set(severity_text, "Warning") where attributes["k8s.event.reason"] == "ProbeWarning"
              or attributes["k8s.event.reason"] == "Unhealthy"
            - set(severity_number, 13) where attributes["k8s.event.reason"] == "ProbeWarning"
              or attributes["k8s.event.reason"] == "Unhealthy"
      receivers:
        k8s_events: null
      service:
        extensions:
        - health_check
        - memory_ballast
        pipelines:
          logs:
            exporters:
            - otlp
            processors:
            - memory_limiter
            - transform/severity
            - transform/namespace
            - transform/entity_attributes
            - resource/events
            - k8sattributes
            - transform/cleanup_attributes_for_nonexisting_entities
            - batch
            receivers:
            - k8s_events
        telemetry:
          logs:
            level: info
          metrics:
            address: 0.0.0.0:8888
